src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -118,26 +117,24 @@ public IRequest getFollowupRequestForCompletion(
 
     @Nullable
     @Override
-    public IRequest onRequestCancelledOrOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    public IRequest<?> onRequestCancelledOrOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
     {
         return null;
     }
 
-    private Set<TileEntityWareHouse> getWareHousesInColony(Colony colony)
+    private static Set<TileEntityWareHouse> getWareHousesInColony(Colony colony)
     {
         return colony.getBuildings().values().stream()
                  .filter(building -> building instanceof BuildingWareHouse)
                  .map(building -> (TileEntityWareHouse) building.getTileEntity())
                  .collect(Collectors.toSet());
     }
 
-    @NotNull
     @Override
     public void onRequestComplete(@NotNull final IToken<?> token)
     {
     }
 
-    @NotNull
     @Override
     public void onRequestCancelled(@NotNull final IToken<?> token)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1186)